docs/executor_plugins.md

```diff
@@ -0,0 +1,105 @@
+# Executor Plugins
+
+## Types
+
+### Template Executor
+
+Run custom ("plugin") templates, e.g for non-pod tasks such as Tekton builds or Spark jobs.
+
+## A Simple Python Plugin
+
+Lets make a Python plugin that prints "hello" each time the workflow is operated on.
+
+We need the following:
+
+1. A HTTP server that will be run as a sidecar to the main container responds to RPC HTTP requests from the executor.
```

Comment:
```suggestion
1. A HTTP server that runs as a sidecar on the step's Pod and responds to RPC HTTP requests from the executor.
```

"sidecar to the main container" seems weird to me, since it's more of a "sidecar to the pod".